scripts/pylib/pytest-twister-harness/src/twister_harness/plugin.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import logging
+import os
+from pathlib import Path
+
+import pytest
+
+from twister_harness.log import configure_logging
+from twister_harness.twister_harness_config import TwisterHarnessConfig
+
+logger = logging.getLogger(__name__)
+
+pytest_plugins = (
+    'twister_harness.fixtures.dut'
+)
+
+
+def pytest_addoption(parser: pytest.Parser):
```

Comment:
Using command line options as the interface between twister core and the harness plugin is not great, this is error prone and depending on  the target and what options it needs for serial capturing and flashing might get too complex and too long. We need to look into a better way of interfacing with the harness other than command line options. Please add this as a future improvement in the tracking issue.